node/tcp/src/helpers/banned_peers.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Aleo Network Foundation
+// This file is part of the snarkOS library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::HashMap, net::IpAddr, time::Instant};
+
+use parking_lot::RwLock;
+
+/// Contains the ban details for a banned peer.
+#[derive(Clone)]
+pub struct BanConfig {
```

Comment:
nit: since there isn't much room for per-ban configuration, `BanDetails` might be more fitting here